packages/beacon-state-transition/src/cache/cachedBeaconState.ts

```diff
@@ -273,6 +274,22 @@ export class BeaconStateContext<T extends allForks.BeaconState> {
     ) as CachedBeaconState<T>;
   }
 
+  getState(): allForks.BeaconState {
+    return this.type.createTreeBacked(this.tree);
+  }
+
+  getNextEpochBeaconProposer(): ValidatorIndex[] {
```

Comment:
This is dangerous because it uses a state, which mutates every slot to mutate data in the epoch context, which is shared between multiple states. `getNextEpochBeaconProposer` should be in the epoch context and use only data available there immutable during an epoch